server.js

```diff
@@ -149,8 +157,96 @@ api.all(/(.*)/, bodyParser.json(), async (req, res) => {
   });
 });
 
+version.use("/:versionNumber", (req, res, next) => {
+  const versionNumber = req.params.versionNumber;
+
+  // Construct the directory path based on the versionNumber path segment
+  const staticDir = path.join(__dirname, "versions", versionNumber);
+
+  // Verify that the file path is under the static directory for security reasons
+  filePath = fs.realpathSync(path.resolve(staticDir));
+  if (!filePath.startsWith(staticDir)) {
+    res.statusCode = 403;
+    res.end();
+    return;
+  } else {
+    if (fs.existsSync(staticDir)) {
+      // Serve static files from the constructed directory
+      express.static(staticDir)(req, res, next);
+    } else {
+      // If the directory doesn't exist, we should check if me can build it or retrieve it ?
+    }
+  }
+});
+
+version.get("/:versionNumber/api", (req, res) => {
+  const versionNumber = req.params.versionNumber;
+
+  const staticDir = path.join(__dirname, "versions", versionNumber);
+
+  // Verify that the file path is under the static directory for security reasons
+  filePath = fs.realpathSync(path.resolve(staticDir));
+  if (!filePath.startsWith(staticDir)) {
+    res.statusCode = 403;
+    res.end();
+    return;
+  } else {
+    const openApiContents = yaml.load(fs.readFileSync(path.join(staticDir, "openapi.yaml")));
+    res.status(200).send(openApiContents);
+  }
```

Comment:
Seems like this block of code is copied three times in the file, maybe it could be factored out

Reply:
Actually, rather see my latest review comment